src/asvo/mod.rs

```diff
@@ -176,112 +177,162 @@ impl AsvoClient {
+                Delivery::Acacia => {
+                    match f.url.as_deref() {
+                        Some(url) => {
+                            debug!("Downloading file {:?}", &url);
+
+                            let op = || {
+                                self.try_download(url, keep_tar, hash, f, job).map_err(Error::transient)
+                            };
+
+                            let _ = retry(ExponentialBackoff::default(), op);
+
+                            info!(
+                                "Completed download in {} (average rate: {}/s)",
+                                if start_time.elapsed().as_secs() > 60 {
+                                    format!(
+                                        "{}min{:.2}s",
+                                        start_time.elapsed().as_secs() / 60,
+                                        (start_time.elapsed().as_millis() as f64 / 1e3) % 60.0
+                                    )
+                                } else {
+                                    format!("{}s", start_time.elapsed().as_millis() as f64 / 1e3)
+                                },
+                                bytesize::ByteSize((total_bytes as u128 * 1000 / start_time.elapsed().as_millis()) as u64)
+                                    .to_string_as(true)
+                            );
+                        },
+                        None => {
+                            return Err(AsvoError::NoUrl {
+                                job_id: job.jobid,
+                            })
+                        }
                     }
-                }
-            } else {
-                // Stream-unzip the response.
-                debug!("Attempting to unzip stream");
-                while let Ok(Some(z)) = read_zipfile_from_stream(&mut tee) {
-                    debug!("Zip file entry: {}", z.name());
-                    if z.is_file() {
-                        debug!("Stream unzipping file {}", z.name());
-                        let mut out_file = File::create(z.name())?;
-                        let mut file_buf = BufReader::with_capacity(buffer_size, z);
-
-                        loop {
-                            let buffer = file_buf.fill_buf()?;
-                            out_file.write_all(buffer)?;
-
-                            let length = buffer.len();
-                            file_buf.consume(length);
-                            if length == 0 {
-                                break;
+                },
+                Delivery::Astro => {
+                    match &f.path {
+                        Some(path) => {
+                            //If it's an /astro job, and the files are reachable from the current host, move them into the current working directory
+                            let path_obj = Path::new(&path);
+                            let folder_name = path_obj.components().last().unwrap().as_os_str().to_str().unwrap();
```

Comment:
Up to you, maybe better for a future release, but you could use the Vec1 package. This provides a data type (vec1) that is the same as a vector, except it's not allowed to be empty. So you could drop unwrap on things like `first()` and `last()`. It's a way of enforcing correctness at the type level.

Reply:
Hey Chris, as far as I can tell, the Path library doesn't provide a way to parse a path into a supplied vector. So we'd need some logic to parse path_obj.components() into a vec1. Doesn't really seem worth doing imo, but will look into it further in a future release.

Reply:
ohhhh it's a `Path`, not a `Vec`. My bad. Never mind!